### DISCUSSION TITLE
Where can I find some useful and comprehensive documentation for Quarto?

### DISCUSSION THREAD

**Discussion by d-lis** [quarto-dev/quarto-cli#11377] [Q&A]
### Description

Which are actually useful and comprehensive, unlike the "guidance" and "reference" on quarto.org?

**Comment by mcanouil** [comment]
That's not really a constructive wording ...
Please be careful of your wording.

What do you mean by "comprehensive"? What is missing exactly and that is not tracked in a GitHub issue?

**Comment by d-lis** [comment]
Indeed, the wording could be more constructive. Please excuse my frustration (which is humongous at this point).

Since I am new to markup and Quatro, coming from a Python background, I have the impression that the documentation is incomplete or patchy and assumes a lot of implicit knowledge from all the resources that are referenced by the docs. From digging through some discussions online, however, it appears to me that I am the only one having trouble to get a grip on this system.

Anyways, comprehensive in this context would mean, for instance, that I would find valid options of document classes right beside the `documentclass` option for pdf documents and not besides `top-level-division`.

That said, let me reformulate my question:
Are there other official resources available, where core functionality of Quarto is documented, besides https://quarto.org/docs ?

**Reply by gadenbuie** [reply]
> I have the impression that the documentation is incomplete or patchy and assumes a lot of implicit knowledge from all the resources that are referenced by the docs.

I can understand this frustration and I think some of it is unfortunately inherent in the broad type of project that is Quarto. At its core, Quarto lets you write documents in a simple markdown format that supports interleaved computations, either as cells or inline in the text content. Writing documents in this way unlocks a multitude of output formats: the same content can be rendered for the web, as a PDF, to a Word document, and so much more. This is a very powerful system that lets you be productive across many many media.

The downside, as you've encountered, is it can be unclear where the boundary between Quarto's features and the "native" feature lies. Your example is a good one: if you're interested in the `documentclass` option for PDF documents, then you're using Quarto to create PDF files via LaTeX. Quarto handles running the computational cells and blending the output with the static portions of your markdown. Then it uses pandoc to convert that markdown into a LaTeX document, where it's finally rendered by LaTeX into the PDF. So Quarto's main job is to coordinate between the `.qmd` format and the output format.

In the case of `documentclass`, this isn't a Quarto feature, _per se_, it's a LaTeX feature, and unfortunately LaTeX [doesn't have a definitive list of document classes, either](https://tex.stackexchange.com/q/782).

In many cases while using Quarto, you generally do not need to know the particulars of the output format you're targeting, if you're okay with the defaults that Quarto chooses. Once you start customizing the output, however, you very often need to transition from learning about Quarto to learning about the underlying technology used in the output.
